cdap-api/src/main/java/co/cask/cdap/api/workflow/AbstractWorkflowAction.java

```diff
@@ -49,7 +50,7 @@ public void destroy() {
     // No-op
   }
 
-  protected final WorkflowContext getContext() {
+  public WorkflowContext getContext() {
```

Comment:
Why becomes public? It should never be. The Workflow driver is the one that pass in the context object to custom action, it shouldn't need to or rely on the action to return it back (since it's user code, it can return something else for you).